andreeapetcu/assignment2/Ancestors.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * ASSUMPTIONS : the binary tree does not contain duplicate values
+ */
+
+import java.util.ArrayList;
+
+public class Ancestors<T> {
+
+  /**
+   * Wraper class that contains the ancestors of a node and a flag for the existence of a node
+   * with the given key.
+   */
+  class Result {
+    public ArrayList<T> ancestors;
+    public boolean flag;
+
+    public Result () {
+      this.ancestors = new ArrayList<>();
+      this.flag = false;
+    }
+  }
+
+  /**
+   * Finds the values of the nodes on the path from the node with the given key to the root.
+   *
+   * @param tree the root of the binary tree
+   * @param key the value of the node whose ancestors are to be printed
+   * @param result wraper class that contains the ancestors and flag for existence of the key
+   * @return wraper class with ancestors of the given key
+   */
+  protected Result printAncestorsAuxiliar(BinaryTreeNode<T> tree, T key, Result result) {
+    if (tree == null) {
+      return result;
+    }
+    
+    // the node with the given value has been found
+    // indicate this by setting the flag to true
+    if ((tree.value).equals(key)) {
```

Comment:
The set of brackets around (tree.value) is unnecessary, and reads a little bit strangely to me. I'd leave them out.